File: apps/mobile/react-native/Environment.ts
import { Environment } from '@khhub/models';

const gatewayPort = 44369;
const authServerPort = 44315;

const apiUrl = `http://localhost:${gatewayPort}`;
const appUrl = 'http://localhost';

const dev: Environment = {
  apiUrl,
  appUrl,
  appName: 'ReactNative',
  oAuthConfig: {
    issuer: `http://localhost:${authServerPort}`,
    clientId: 'ReactNative',
    scope:
      'offline_access AuthServer IdentityService AdministrationService',
  },
  localization: {
    defaultResourceName: 'KHHub',
  },
};

const prod: Environment = { ...dev };

export const getEnvVars = () => (__DEV__ ? dev : prod);

File: apps/mobile/khub_flutter_app/lib/src/core/config/environment.dart
class OAuthConfig {
  const OAuthConfig({
    required this.issuer,
    required this.clientId,
    required this.scope,
    this.clientSecret,
  });

  final String issuer;
  final String clientId;
  final String scope;
  final String? clientSecret;
}

class Environment {
  const Environment({
    required this.apiUrl,
    required this.appUrl,
    required this.appName,
    required this.oAuthConfig,
    required this.defaultResourceName,
  });

  final String apiUrl;
  final String appUrl;
  final String appName;
  final OAuthConfig oAuthConfig;
  final String defaultResourceName;
}

const Environment currentEnv = Environment(
  apiUrl: 'https://api.khub.id.vn',
  appUrl: 'https://khub.id.vn',
  appName: 'ReactNative',
  oAuthConfig: OAuthConfig(
    issuer: 'https://auth.khub.id.vn',
    clientId: 'ReactNative',
    scope: 'offline_access AuthServer IdentityService AdministrationService',
  ),
  defaultResourceName: 'KHHub',
);
